sockeye/data_io.py

```diff
@@ -257,9 +281,13 @@ def __init__(self,
         num_buckets = len(buckets)
         self.length_ratio_mean = length_ratio_mean
         self.length_ratio_std = length_ratio_std
-        self.unk_id_source = vocab_source[C.UNK_SYMBOL]
+        if vocab_source is not None:
```

Comment:
If this can be None, the typing annotation in the constructor should reflect this: `Optional[Dict[str, int]]`.

Reply:
Ok